presto-main/src/main/java/com/facebook/presto/sql/relational/Signatures.java

```diff
@@ -155,6 +156,11 @@ public static Signature inSignature()
         return internalScalarFunction(IN, StandardTypes.BOOLEAN);
     }
 
+    public static Signature rowConstructorSignature(Type returnType, List<Type> argumentTypes)
+    {
+        return internalScalarFunction(ROW_CONSTRUCTOR, returnType.getTypeSignature(), Lists.transform(argumentTypes, Type::getTypeSignature));
```

Comment:
We're mostly moving away from `Lists.transform`, so use `argumentTypes.stream().map(Type::getTypeSignature).collect(toImmutableList())`